### ISSUE TITLE
current_timestamp breaks json_read_auto via stdin

### ISSUE THREAD

**Issue by benjiman** [duckdb/duckdb#11041]
### What happens?

current_timestamp/today seems to break json schema inference when tables with json and current_timestamp are joined together, but only when reading from standard input.

took a while to narrow this down to a small test case but see the to-reproduce for the simplest example I could produce.  

### To Reproduce

```
# simple json file
$:/tmp> cat example2.json 
[{"foo":"bar"},{"foo":"baz"}] 

# querying fine with literal timestamps

$:/tmp> cat example2.json | duckdb -c "select foo from read_json_auto('/dev/stdin',format='array') union select '2024-03-07'::timestamp::varchar as foo "
┌─────────────────────┐
│         foo         │
│       varchar       │
├─────────────────────┤
│ baz                 │
│ 2024-03-07 00:00:00 │
│ bar                 │
└─────────────────────┘


# querying fails when current_timestamp involved
$:/tmp> cat example2.json | duckdb -c "select foo from read_json_auto('/dev/stdin',format='array') union select current_timestamp::varchar as foo "
Error: Binder Error: Referenced column "foo" not found in FROM clause!
Candidate bindings: "read_json_auto.json"
LINE 1: select foo from read_json_auto('/dev/stdin',fo...
               ^

# same query works fine when reading directly from file rather than via stdin
$:/tmp> duckdb -c "select foo from read_json_auto('/tmp/example2.json',format='array') union select current_timestamp::varchar as foo "
┌────────────────────────────┐
│            foo             │
│          varchar           │
├────────────────────────────┤
│ baz                        │
│ bar                        │
│ 2024-03-07 11:39:27.433+00 │
└────────────────────────────┘
```

today() has the same issue, although interestingly age() does not which I'm using a workaround

### OS:

Linux x64

### DuckDB Version:

v0.10.0 20b1486d11

### DuckDB Client:

binary

### Full Name:

Benji Weber

### Affiliation:

Mindgard

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Thanks, we'll take a look.

**Comment by benjiman** [comment]
This is still an issue. Having to work around it.

**Comment by benjiman** [comment]
This is still a problem with current nightly v1.1.3-dev156 9cba6a2a03 and still rreproducable

**Comment by benjiman** [comment]
Confirming this is fixed in v1.3.0-dev1180 5f7d3e02dd (not sure exactly which version resolved it)
